pandas/core/groupby/grouper.py

```diff
@@ -585,8 +585,13 @@ def _make_codes(self) -> None:
                 codes = self.grouper.codes_info
                 uniques = self.grouper.result_index
             else:
+                # GH35667, replace dropna=False with na_sentinel=None
+                if not self.dropna:
+                    na_sentinel = None
+                else:
+                    na_sentinel = -1
                 codes, uniques = algorithms.factorize(
-                    self.grouper, sort=self.sort, dropna=self.dropna
+                    self.grouper, sort=self.sort, na_sentinel=na_sentinel
```

Comment:
Could maybe also be `na_sentinel=-1 if self.dropna else None` (if that fits on the line)

Reply:
yeha, i tried this already, but black would split it to multi-line since it is 90 words + long

do you still prefer to have this? i could also change this way.

Reply:
In that case fine as it is!